temporal-sdk/src/main/java/io/temporal/workflow/UpdateValidateMethod.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.workflow;
+
+import io.temporal.common.Experimental;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.annotation.Nonnull;
+
+/**
+ * Indicates that the method is an update validator handle. An update validator handle is associated
+ * with an update method and runs before the associated update handle. If the update validator
+ * throws an exception, the update handle is not called and the update is not persisted in history.
+ *
+ * <p>This annotation applies only to workflow interface methods.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+@Experimental
+public @interface UpdateValidateMethod {
+
+  /**
+   * Name of the update handler the validator should be used for. Must not be null.
+   *
+   * <p>Be careful about names that contain special characters. These names can be used as metric
+   * tags. And systems like prometheus ignore metrics which have tags with unsupported characters.
+   */
+  @Nonnull
+  String updateName();
```

Comment:
The other option I thought of was to have the update declare the validator it wanted, more similar to Go

Reply:
This is better IMO since it's often the case an update may not have a validator, it's best to have the validator attach itself vs the update attach itself to the validator.

Reply:
Only thing is that I don't think this could support reusing validators  for multiple update functions, could it?

Reply:
correct that is a drawback @mmcshane

Reply:
I think that's an acceptable drawback, same as not reusing signal handlers or update handlers for multiple purposes.

Reply:
That being the case, I'd throw my vote with specifying the validator method from the update annotation with a default validator of "__unused" or something like that. It's a mild preference and I would defer to the sdk owner.